src/helpers/guards/isNonEmptyArray.ts

```diff
@@ -0,0 +1,3 @@
+export function isNonEmptyArray<T>(array: Array<T>): array is NonEmptyArray<T> {
```

Comment:
I'm going to be honest I don't super love the fact that we are writing code for this when an empty array could just be tested with a !!exampleArray as it would return false if the array is empty. I could be wrong and maybe this function serves a larger purpose but it does feel a little wrong.

Reply:
To be clear, you're absolutely right that this function shouldn't be necessary, but I've found it helpful in enough spots that I brought it in here.